server/src/animations/animations.service.ts

```diff
@@ -1,35 +1,84 @@
 import { Injectable } from '@nestjs/common';
 import { Animation, Prisma } from '@prisma/client';
 import { PrismaService } from 'src/prisma/prisma.service';
+import { CacheService } from 'src/cache/cache.service';
 
 @Injectable()
 export class AnimationsService {
-  constructor(private readonly prisma: PrismaService) {}
+  constructor(
+    private readonly prisma: PrismaService,
+    private readonly cacheService: CacheService,
+  ) {}
 
-  create(data: Prisma.AnimationCreateInput): Promise<Animation> {
-    return this.prisma.animation.create({ data });
+  createCacheKey(key: string | number) {
+    return `${CacheService.KEY_ANIMATION}_${key}`;
   }
 
-  getAnimationById(id: number): Promise<Animation | null> {
-    return this.prisma.animation.findUnique({
-      where: { id },
-      include: {
-        Scene: true,
-      },
-    });
+  async resetAnimationCache(id: string | number) {
+    const cacheKey = this.createCacheKey(id);
+    return await this.cacheService.reset(cacheKey);
+  }
+
+  async create(data: Prisma.AnimationCreateInput): Promise<Animation> {
```

Comment:
Normally we'll try to abstract the fact we're using Prisma etc at the service data contact level. Not a big deal, but really in either the same way you're using the DTOs. This approach does work well with the prisma generate however.